exasol_integration_test_docker_environment/lib/docker/images/create/utils/file_directory_list_hasher.py

```diff
@@ -56,50 +82,88 @@ def __init__(self,
             self.excluded_extensions = []
 
         self.path_hasher = PathHasher(hashfunc,
-                                      hash_permissions=hash_permissions,
-                                      use_relative_paths=use_relative_paths)
+                                      hash_permissions=hash_permissions)
         self.file_content_hasher = FileContentHasher(hashfunc, blocksize)
 
-    def hash(self, files_and_directories: List[str]) -> bytes:
-        collected_directories = []
-        collected_files = []
+    def hash(self, files_and_directories: List[PathMapping]) -> bytes:
         if not isinstance(files_and_directories, List):
             raise Exception("List with paths expected and not '%s' with type %s"
                             % (files_and_directories, type(files_and_directories)))
+
+        directory_mapping_result = self.collect_dest_path_and_src_files(files_and_directories)
+        hashes = self.compute_hashes(directory_mapping_result)
+        return self._reduce_hash(hashes)
+
+    def validate(self, mappings: List[DestinationMapping]) -> None:
+        # Verify that there are no duplicate mappings to same destination
+        destinations = [p.destination for p in mappings]
+        if len(set(destinations)) != len(destinations):
+            raise AssertionError(f"Directory content for hashing contains duplicates: {destinations}")
+
+    def collect_dest_path_and_src_files(self, files_and_directories: List[PathMapping]) -> DirectoryMappingResult:
+        """
+        Iterate over all mappings and collect:
+        1) Destination paths-names (for Hashing): Mapping of source -> destination of file/directory, ordered by destination path
+        2) Source files, ordered by destination path
+        Collection is done considering the current configuration
+         - for both: excluded_directory/excluded_extensions/excluded_file
+         - for 1): hash_file_names/hash_directory_names
+        """
+        collected_dest_paths: List[DestinationMapping] = list()
+
+        def replace_src_by_dest_path(src: str, dest: str, target: str) -> str:
+            if not target.startswith(src):
```

Comment:
Path is_relative_to

Reply:
This function is only available since Python 3.9 ( i had it already tried)